tests/PSRule.Rules.Azure.Tests/Azure.Storage.Tests.ps1

```diff
@@ -199,6 +199,34 @@ Describe 'Azure.Storage' -Tag Storage {
             $ruleResult.Length | Should -Be 1;
             $ruleResult.TargetName | Should -BeIn 'storage-D';
         }
+
+        It 'Azure.Storage.FileShareSoftDelete' {
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.Storage.FileShareSoftDelete' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetName | Should -BeIn 'storage-F';
+
+            $ruleResult[0].Reason | Should -Not -BeNullOrEmpty;
+            $ruleResult[0].Reason | Should -BeExactly 'A sub-resource of type ''Microsoft.Storage/storageAccounts/fileServices'' has not been specified.';
+
+            # Pass
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetName | Should -BeIn 'storage-G';
+            $ruleResult[0].Reason | Should -Not -BeNullOrEmpty;
+            $ruleResult[0].Reason | Should -BeExactly 'Path properties.shareDeleteRetentionPolicy.enabled: Is set to ''True''.';
```

Comment:
@jonathanruiz You need to add a passing test case into tests/PSRule.Rules.Azure.Tests/Resources.Storage.json. Maybe edit storage-G to include configure a `shareDeleteRetentionPolicy` with `enabled` and `days` set.